### PR TITLE
Added shell_sort.py

### LINKED ISSUE #192: Shell Sort
I would like to implement Shell Sort algorithm as part of GSSOC'21. Kindly assign me this issue. @Joshitha18 @rcoder23

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
Fixes #192 

@Joshitha18  @rcoder23  Kindly review my PR.

### CODE REVIEW

**Review by rohanbabbar04 — APPROVED** [review]

File: Algorithms/Sorting/shell_sort.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Shell sort is an improvised version of insertion sort, where we move
+# elements by more than one positions
+def shellSort(arr, n):
+
+    gap = n // 2
+
+    while(gap > 0):
+        for i in range(gap, n):
+            ele = arr[i]
```
Comment (rcoder23) [inline Algorithms/Sorting/shell_sort.py:9]: Producing Error **list index out of range**

**Review by rcoder23 — CHANGES_REQUESTED** [review]
Recheck your code .

**Comment by Harikrishnan6336** [comment]
I have tested the code with various edge case sample input outputs and then also wasnt able to find an occurence of such error!!.
Could you tell me for which test case the error is occuring. @rcoder23

File: Algorithms/Sorting/shell_sort.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                arr[index] = arr[index-gap]
+                index -= gap
+            arr[index] = ele
+        gap //= 2
+
+
+if __name__ == '__main__':
+    print("How many numbers do you want to sort? ", end="")
+    n = int(input())
+        if n == 0:
```
Comment (rohanbabbar04) [inline Algorithms/Sorting/shell_sort.py:21]: Unexpected Indent

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]

**Comment by Harikrishnan6336** [comment]
Made the required changes.

**Review by rcoder23 — APPROVED** [review]
